File: April-18/24-04-18/python/01-Password.py
def pali(str1, str2):
    l1 = len(str1)
    l2 = len(str1)
    if(l1 != l2):
        return False
    
    for i in range(0, l1):
        if(str1[i] != str2[l2-i-1]):
            return False
    return True

n = int(input())
string = [input() for x in range(0,n)]

l = 0
ch = ' '
for i in range(0, n):
    for j in range(i+1, n):
        if(pali(string[i], string[j])):
            l = len(string[i])
            ch = string[i][l//2]
            break
print(l, ch)

File: April-18/24-04-18/java/01-Password.java

import java.io.*;


class TestClass {
    static boolean pali(String str1, String str2)
    {
        int l1 = str1.length();
        int l2 = str1.length();
        if(l1 != l2)
            return false;
            
        for(int i = 0; i < l1; i++)
        {
            if(str1.charAt(i) != str2.charAt(l2-i-1))
            {
                return false;
            }
        }
        return true;
    }
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(input.readLine());
        String str[] = new String[n];
        for(int i = 0; i < n; i++)
        {
            str[i] = input.readLine();
        }
        
        int len = 0;
        char ch = ' ';
        
        for(int i = 0; i < n; i++)
        {
            for(int j = i + 1; j < n; j++)
            {
                if(pali(str[i],str[j]))
                {
                    len = str[i].length();
                    ch = str[i].charAt(len/2);
                    break;
                }
            }
        }
        System.out.println(len + " " + ch);
    }
}
